plugins/com.google.cloud.tools.eclipse.appengine.newproject.test/src/com/google/cloud/tools/eclipse/appengine/newproject/Maven4NamespaceContext.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2018 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.eclipse.appengine.newproject;
+
+import com.google.common.collect.ImmutableList;
+import java.util.Iterator;
+import javax.xml.namespace.NamespaceContext;
+
+/**
+ * Maps all prefixes to http://maven.apache.org/POM/4.0.0 and
+ * all namespace URIs to "m". Not suitable for general use.
+ */
+class Maven4NamespaceContext implements NamespaceContext {
```

Comment:
You can reuse `com.google.cloud.tools.eclipse.appengine.validation.MavenContext`

Reply:
I'd forgotten about that class. Looking at it now, it has the same issues this class does. That is, it works only for very limited code and is actively broken for all other uses. Given that, I don't feels safe exposing it as part of the public API. We could look for or build a more correct namespace context that fully satisfies the interface's contract.